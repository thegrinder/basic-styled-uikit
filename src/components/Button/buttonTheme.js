import { rem } from '../../helpers/utils';

const buttonTheme = {
  btnTypes: {
    default: {
      normal: {
        color: '#222',
        backgroundColor: 'transparent',
        borderColor: '#e5e5e5',
      },
      hover: {
        color: '#222',
        backgroundColor: 'transparent',
        borderColor: '#b2b2b2',
      },
      active: {
        color: '#222',
        backgroundColor: 'transparent',
        borderColor: '#999',
      },
    },
    primary: {
      normal: {
        color: '#fff',
        backgroundColor: '#1e87f0',
        borderColor: 'transparent',
      },
      hover: {
        color: '#fff',
        backgroundColor: '#0f7ae5',
        borderColor: 'transparent',
      },
      active: {
        color: '#fff',
        backgroundColor: '#0e6dcd',
        borderColor: 'transparent',
      },
    },
    secondary: {
      normal: {
        color: '#fff',
        backgroundColor: '#222',
        borderColor: 'transparent',
      },
      hover: {
        color: '#fff',
        backgroundColor: '#151515',
        borderColor: 'transparent',
      },
      active: {
        color: '#fff',
        backgroundColor: '#080808',
        borderColor: 'transparent',
      },
    },
    danger: {
      normal: {
        color: '#fff',
        backgroundColor: '#f0506e',
        borderColor: 'transparent',
      },
      hover: {
        color: '#fff',
        backgroundColor: '#ee395b',
        borderColor: 'transparent',
      },
      active: {
        color: '#fff',
        backgroundColor: '#ec2147',
        borderColor: 'transparent',
      },
    },
  },
  btnSizes: {
    small: {
      padding: `0 ${rem(15)}`,
      lineHeight: rem(28),
      fontSize: rem(10),
    },
    default: {
      padding: `0 ${rem(30)}`,
      fontSize: rem(12),
      lineHeight: rem(38),
    },
    large: {
      padding: `0 ${rem(40)}`,
      lineHeight: rem(53),
      fontSize: rem(14),
    },
  },
};

export default buttonTheme;
