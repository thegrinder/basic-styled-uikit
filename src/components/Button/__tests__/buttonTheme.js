import { buttonBaseTheme, buttonDarkTheme } from '../buttonTheme';

describe('buttonBaseTheme', () => {
  it('should be defined', () => {
    expect(buttonBaseTheme).toBeDefined();
  });
});

describe('buttonDarkTheme', () => {
  it('should be defined', () => {
    expect(buttonDarkTheme).toBeDefined();
  });
});
