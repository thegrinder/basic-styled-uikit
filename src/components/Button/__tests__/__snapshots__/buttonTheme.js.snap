// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buttonTheme should be defined 1`] = `
Object {
  "btnTypes": Object {
    "danger": Object {
      "active": Object {
        "backgroundColor": "#bc3409",
      },
      "hover": Object {
        "backgroundColor": "#d53b0a",
      },
      "normal": Object {
        "backgroundColor": "#F4511E",
        "color": "#fff",
      },
    },
    "default": Object {
      "active": Object {
        "backgroundColor": "#60686f",
      },
      "hover": Object {
        "backgroundColor": "#6c757d",
      },
      "normal": Object {
        "backgroundColor": "#868e96",
        "color": "#fff",
      },
    },
    "primary": Object {
      "active": Object {
        "backgroundColor": "#1360a4",
      },
      "hover": Object {
        "backgroundColor": "#166dba",
      },
      "normal": Object {
        "backgroundColor": "#1E88E5",
        "color": "#fff",
      },
    },
  },
  "common": Object {
    "fontFamily": "Lato, Arial, Helvetica, sans-serif",
    "fontWeight": "700",
  },
  "sizings": Object {
    "l": Object {
      "fontSize": "1.375rem",
      "lineHeight": "3.375rem",
      "padding": "0 2.5rem",
    },
    "m": Object {
      "fontSize": "1rem",
      "lineHeight": "2.375rem",
      "padding": "0 1.875rem",
    },
    "s": Object {
      "fontSize": ".875rem",
      "lineHeight": "1.75rem",
      "padding": "0 1rem",
    },
  },
}
`;
